tests/test_lists2safebrowsing.py

```diff
@@ -168,3 +215,279 @@ def test_add_domain_to_list_duplicate():
     assert not added
     assert output == [domain_hash.digest()]
     assert not log_writes
+
+
+@pytest.fixture
+def chunknum():
+    return int(time.time())
+
+
+def _write_safebrowsing_blocklist(chunknum, version=None,
+                                  add_test_domains=True,
+                                  write_to_file=True):
+    """Auxiliary function for write_safebrowsing_blocklist tests."""
+    domains = [d[1] for d in CANONICALIZE_TESTCASES]
+    canonicalized_domains = [d[2] for d in CANONICALIZE_TESTCASES]
+    output_name = "test-track-digest256"
+    section = "tracking-protection-test"
+
+    with patch("test_lists2safebrowsing.open", mock_open()):
+        with open(output_name, "wb") as output_file:
+            output_file = output_file if write_to_file else None
+            l2s.write_safebrowsing_blocklist(domains, output_name, None,
+                                             chunknum, output_file,
+                                             section, version)
+
+    output_writes = output_file.write.call_args_list if write_to_file else []
+
+    expected_domains = []
+    if add_test_domains:
+        expected_domains.append(TEST_DOMAIN_TEMPLATE % output_name + "/")
+        if version:
+            expected_domains.append("%s-%s" % (version.replace(".", "-"),
+                                    expected_domains[0]))
+
+    # Remove duplicates while preserving order
+    expected_domains += list(OrderedDict.fromkeys(canonicalized_domains))
+    expected_hashes = [hashlib.sha256(d.encode("utf-8")).digest()
+                       for d in expected_domains]
+    expected_bytes = hashlib.sha256().digest_size * len(expected_hashes)
+    expected_header = b"a:%d:32:%d\n" % (chunknum, expected_bytes)
+    expected_output = expected_header + b"".join(expected_hashes)
+    expected_print = (PRINT_MSG % ("Tracking protection", section,
+                      len(expected_domains), len(expected_output)))
+
+    return output_writes, expected_output, expected_print
```

Comment:
Please do not create the expected values programmatically as it causes confusion on what the expected value should be and needs to be maintained as it is duplicated logic from the code we are testing.